core/synch.c

```diff
@@ -1749,6 +1749,10 @@ translate_from_synchall_to_dispatch(thread_record_t *tr, thread_synch_state_t sy
                 arch_mcontext_reset_stolen_reg(dcontext, mc);
             }
         });
+        IF_AARCHXX({
+            set_stolen_reg_val(mc, (reg_t)os_get_dr_tls_base(dcontext));
```

Comment:
Do we  need to save the existing value of the stolen reg somehow?

Reply:
A client xl8 could have restored a value there so it seems yes.  (I wonder if the signal redirect does that properly.)

Reply:
`transfer_from_sig_handler_to_fcache_return` doesn't seem to save the existing stolen reg value before writing it.
https://github.com/DynamoRIO/dynamorio/blob/6dea75659134018d0796e7c923b62db03fcc6cf2/core/unix/signal.c#L3426

Is it because the spill slot will continue to have that value, and can be restored later?
https://github.com/DynamoRIO/dynamorio/blob/6dea75659134018d0796e7c923b62db03fcc6cf2/core/translate.c#L1062

Reply:
No, the client xl8 cb is called during xl8 and its result is not kept anywhere else.  It is likely rare for a client to mess w/ the app's stolen reg (most tools are observational only; drreg avoids using the stolen reg; etc.) but it seems like a real theoretical issue to me.  How about we ignore it here and match signal.c and I file an issue on it and we fix it in both places later if we all agree about it.

I filed #4495.

Reply:
If you could add a XXX or TODO comment about #4495.

Reply:
Done.

Reply:
Done.